endaq/calc/stats.py

```diff
@@ -6,6 +6,342 @@
+    :param zero: the output quantity driven to zero prior to the calculations
+        `None` does nothing
+        `"start"` forces the first datapoint to 0,
+        `"mean"` chooses ``-np.mean(output)`` & `"median"` (default) chooses
+        ``-np.median(output)``
+    :param highpass_cutoff: the cutoff frequency of a preconditioning highpass
+        filter; if None, no filter is applied. For shock events, it is recommended to set this to None (the default), but it is recommended for vibration.
+    :param vel_disp_multiplier: applies a scale to the velocity and displacement metrics
+        - 386.09 to convert from g to inches (in)
+        - 9806.65 to convert from g to millimeters (mm)
+    :param include_integration: include the calculations of velocity and displacement, default behavior
+    :param include_pseudo_velocity: include the more time consuming calculation of pseudo velocity, default behavior
+    :param damp: the damping coefficient `ζ`, related to the Q-factor by
+        `ζ = 1/(2Q)`; defaults to 0
+    :param init_freq: the initial frequency in the sequence; if `None`,
+        use the frequency corresponding to the data's duration, default is 1.0 Hz
+    :param bins_per_octave: the number of frequencies per octave
+    :param include_resultant: add a resultant (root sum of the squares) for each metric, calculated from the other input dataframe columns
+    :param display_plots: display plotly figures of the min/max envelope of acceleration, velocity, displacement and PVSS (default as False)
+    :return: a dataframe containing all the metrics, one computed per column of the input dataframe
+    """
+
+    #Remove Offset
+    if zero is not None:
+        if zero == "start":
+            df = df - df.iloc[0]
+        elif zero in ("mean", "median"):
+            zero = dict(mean=np.mean, median=np.median)[zero]
+            df = df - df.apply(zero, axis="index", raw=True)
+
+    #Apply Filters & Window
+    df = filters.butterworth(df, low_cutoff=highpass_cutoff, tukey_percent=tukey_percent)
+
+    #Integrate
+    data_list = [df]
+    if include_integration:
+        [accel, vel, disp] = integrate.integrals(
+            df, n=2, tukey_percent=tukey_percent, highpass_cutoff=highpass_cutoff)
+        data_list = [df, vel, disp]
+
+    #Calculate Peak & RMS
+    metrics = pd.DataFrame()
+    rms_start = int(tukey_percent/2*df.shape[0])
+    rms_end = df.shape[0] - rms_start
+    for (label,data,scale) in zip(
+            ['Acceleration','Velocity','Displacement'],
+            data_list,
+            [1,vel_disp_multiplier,vel_disp_multiplier]):
+
+        #Apply Scale
+        data = data * scale
+
+        #Display Plots
+        if display_plots:
+            rolling_min_max_envelope(data).update_layout(yaxis_title_text = label).show()
+
+        #Calculate Absolute Peak
+        peak = pd.DataFrame(data.abs().max()).reset_index()
+        peak.columns = ['variable','value']
```

Comment:
Not going to make comments on every single one, but values separated by commas should also be followed by a space.  
```suggestion
        peak.columns = ['variable', 'value']
```